### PR TITLE
[GO] Fixing compilation issue when pathParams  is not presented

### PR DESCRIPTION
**Pull request by guohuang**
@wing328 please review, this is solving the `pathParams ` empty issue

### CODE REVIEW

File: modules/swagger-codegen/src/main/java/io/swagger/codegen/languages/GoClientCodegen.java [hunk 1]
```diff
@@ -382,13 +382,21 @@ public String toOperationId(String operationId) {
...
-                imports.add(customImport);
+                imports.add(createMapping("import", "encoding/json"));
                 break; //just need to import once
             }
         }
 
+        // this will only import "strings" "fmt" if there are items in pathParams
+        for (CodegenOperation operation : operations) {
+            if(operation.pathParams != null && operation.pathParams.size() > 0) {
+                imports.add(createMapping("import", "fmt"));
```
Comment (JackDanger) [inline modules/swagger-codegen/src/main/java/io/swagger/codegen/languages/GoClientCodegen.java]: What a good idea!

**Comment by wing328** [comment]
CI reported failure running Go tests:
```
# _/home/travis/build/swagger-api/swagger-codegen/samples/client/petstore/go/go-petstore
go-petstore/pet_api.go:26: imported and not used: "errors"
go-petstore/store_api.go:26: imported and not used: "errors"
go-petstore/user_api.go:26: imported and not used: "errors"
FAIL	_/home/travis/build/swagger-api/swagger-codegen/samples/client/petstore/go [build failed]
```

Please take a look when you've time.

File: modules/swagger-codegen/src/main/java/io/swagger/codegen/languages/GoClientCodegen.java (hunk 1 above)
Comment (wing328) [inline modules/swagger-codegen/src/main/java/io/swagger/codegen/languages/GoClientCodegen.java]: The `postProcessOperations` and `postProcessModels` are pretty useful in meeting language-specified requirement.

**Comment by guohuang** [comment]
@wing328 sorry, I accidentally added the "errors" reference, it is fixed now

**Comment by wing328** [comment]
@guohuang no worry. The CI caught it :)

I'll merge it shortly. Have a nice weekend.

**Pull request merged**